src/utils/read-rows-resumption.ts

```diff
@@ -0,0 +1,270 @@
+// Copyright 2024 Google LLC
```

Comment:
This file largely contains the retry logic that was in createreadstream before. Most of it has been moved over as is.